multiversxetl/cli/work.py

```diff
@@ -206,25 +201,54 @@ def do_any_load_task(
         file_storage.remove_transformed_file(task.get_pretty_name())
 
         logger.log_info(f"Transform & load finished, index = {task.index_name}, task = {task.id}", data=task.to_dict())
-    except KeyboardInterrupt:
-        storage.update_task(task.id, lambda t: t.update_on_extraction_failure(ERROR_INTERRUPTED_TASK))
-        logger.log_error(f"Transform & load interrupted, index = {task.index_name}, task = {task.id}", data=task.to_dict())
-        raise
     except Exception as e:
         storage.update_task(task.id, lambda t: t.update_on_loading_failure(str(e)))
         logger.log_error(f"Transform & load failed, index = {task.index_name}, task = {task.id}", data=task.to_dict())
         raise
 
 
-def do_continuously(callable: Callable[[], None], continue_on_error: bool, sleep_between_tasks: int):
-    while True:
+def do_continuously(callable: Callable[[], None], sleep_between_tasks: int, num_threads: int, thread_start_delay: int = 1):
+    threads: List[threading.Thread] = []
+    should_stop_all_threads = threading.Event()
+
+    for i in range(num_threads):
+        # Start threads with a small delay between them.
+        time.sleep(thread_start_delay)
+
+        thread = threading.Thread(
+            name=f"Thread-{i}",
+            target=do_continuously_in_thread,
+            args=[callable, sleep_between_tasks, should_stop_all_threads]
+        )
+
+        thread.start()
+        threads.append(thread)
+
+        if should_stop_all_threads.is_set():
+            break
+
+    for thread in threads:
+        if thread.is_alive():
+            thread.join()
+
+
+def do_continuously_in_thread(callable: Callable[[], None], sleep_between_tasks: int, should_stop: threading.Event):
+    logging.info(f"Thread started.")
```

Comment:
maybe you can log also the thread name?

Reply:
:+1:

Reply:
Thread name is visible in the formatted logs, as follows:

```
[Thread-0] [INFO] [work]: Thread started.
[Thread-1] [INFO] [work]: Thread started.
[Thread-0] [INFO] [work]: No tasks left, try again later.
[Thread-0] [INFO] [work]: Sleeping for 3 seconds...
```

Due to: https://github.com/multiversx/multiversx-etl/blob/main/multiversxetl/__main__.py#L5